frontend/public/components/secrets/create-secret/PullSecretCredentialEntry.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+export const PullSecretCredentialEntry: React.FC<PullSecretCredentialEntryProps> = ({
+  id,
+  address,
+  email,
+  password,
+  username,
+  onChange,
+}) => {
+  const { t } = useTranslation();
+
+  const updateEntry = (name, value) =>
+    onChange({ address, username, password, email, [name]: value }, id);
```

Comment:
Sorry, my example didn't include types, but we should have type definitions on this:
```suggestion
  const updateEntry = (name: string, value: string): void =>
    onChange({ address, username, password, email, [name]: value }, id);
```